### PR TITLE
[flutter_tools] switch web integration tests to use html

### PR DESCRIPTION
**Pull request by jonahwilliams**
canvaskit is pulled from the net and may require more work than html for these test bots

Need to add loops because of https://github.com/flutter/flutter/issues/86202

### CODE REVIEW

**Comment by zanderso** [comment]
It looks like the html version of the hot restart test is failing with a timeout in the presubmit checks.

**Comment by jonahwilliams** [comment]
gosh darn it

**Comment by zanderso** [comment]
> gosh darn it

If it fails more consistently now, that is progress.

**Comment by jonahwilliams** [comment]
Yup, fairly consistent now - at least I can debug this locally!

**Comment by jonahwilliams** [comment]
these tests should be passing now, besides customer tests being broken

**Comment by jonahwilliams** [comment]
@zanderso PTAL

File: packages/flutter_tools/test/integration.shard/test_data/hot_reload_project.dart [hunk 1]
```diff
@@ -25,12 +25,21 @@ class HotReloadProject extends Project {
...
+  import 'package:flutter/foundation.dart';
 
   void main() async {
     WidgetsFlutterBinding.ensureInitialized();
     final ByteData message = const StringCodec().encodeMessage('AppLifecycleState.resumed')!;
     await ServicesBinding.instance!.defaultBinaryMessenger.handlePlatformMessage('flutter/lifecycle', message, (_) { });
-    runApp(MyApp());
+    // See https://github.com/flutter/flutter/issues/86202
+    if (kIsWeb) {
+     while (true) {
```
Comment (zanderso) [inline packages/flutter_tools/test/integration.shard/test_data/hot_reload_project.dart:36]: nit: check indentation.

File: packages/flutter_tools/test/integration.shard/test_data/hot_reload_project.dart [hunk 2]
```diff
@@ -25,12 +25,21 @@ class HotReloadProject extends Project {
...
   import 'package:flutter/services.dart';
   import 'package:flutter/widgets.dart';
+  import 'package:flutter/foundation.dart';
 
   void main() async {
     WidgetsFlutterBinding.ensureInitialized();
     final ByteData message = const StringCodec().encodeMessage('AppLifecycleState.resumed')!;
     await ServicesBinding.instance!.defaultBinaryMessenger.handlePlatformMessage('flutter/lifecycle', message, (_) { });
-    runApp(MyApp());
+    // See https://github.com/flutter/flutter/issues/86202
```
Comment (zanderso) [inline packages/flutter_tools/test/integration.shard/test_data/hot_reload_project.dart:34]: To make sure I understand, the solution to prints getting dropped that is being used here is to just call `runApp` repeatedly? When does it stop?

Comment (jonahwilliams) [reply packages/flutter_tools/test/integration.shard/test_data/hot_reload_project.dart:34]: yeah, run app causes a rebuild which triggers more prints. It stops when the app is stopped

File: packages/flutter_tools/test/web.shard/expression_evaluation_web_test.dart [hunk 3]
```diff
@@ -170,8 +170,7 @@ void main() {
       await startPaused(expressionEvaluation: true);
       await evaluateComplexExpressionsInLibrary(flutter);
     });
-  }, skip: true, // Flaky tests: https://github.com/flutter/flutter/issues/84012
-  );
+  });
```
Comment (annagrin) [inline packages/flutter_tools/test/web.shard/expression_evaluation_web_test.dart:173]: is it possible to unskip the other test as well (line 107 above)?

File: packages/flutter_tools/test/web.shard/debugger_stepping_web_test.dart [hunk 4]
```diff
@@ -27,7 +27,7 @@ void main() {
 
     await flutter.run(
       withDebugger: true, startPaused: true, chrome: true,
-      additionalCommandArgs: <String>['--verbose']);
+      additionalCommandArgs: <String>['--verbose', '--web-renderer=html']);
```
Comment (annagrin) [inline packages/flutter_tools/test/web.shard/debugger_stepping_web_test.dart:30]: can we unskip the test on line 51?

File: packages/flutter_tools/test/web.shard/hot_reload_web_test.dart [hunk 5]
```diff
@@ -32,7 +32,7 @@ void main() {
   });
 
   testWithoutContext('hot restart works without error', () async {
-    await flutter.run(chrome: true, additionalCommandArgs: <String>['--verbose']);
+    await flutter.run(chrome: true, additionalCommandArgs: <String>['--verbose', '--web-renderer=html']);
```
Comment (annagrin) [inline packages/flutter_tools/test/web.shard/hot_reload_web_test.dart:35]: can we unskip the tests on lines 54 and 73?

**Review by annagrin — COMMENTED** [review]
LGTM

**Review by annagrin — APPROVED** [review]

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Lets see how the other skipped tests do

**Pull request merged**